pimd/pim_addr.h

```diff
@@ -21,19 +21,88 @@
 #define _PIMD_PIM_ADDR_H
 
 #include "jhash.h"
+#include "prefix.h"
+
+/* clang-format off */
 
 /* temporarily disable IPv6 types to keep code compiling.
  * Defining PIM_V6_TEMP_BREAK will show a lot of compile errors - they are
  * very useful to see TODOs.
  */
 #if PIM_IPV == 4 || !defined(PIM_V6_TEMP_BREAK)
 typedef struct in_addr pim_addr;
-#define PIM_ADDRSTRLEN INET_ADDRSTRLEN
+
+#define PIM_ADDRSTRLEN	INET_ADDRSTRLEN
+#define PIM_AF		AF_INET
+#define PIM_AFI		AFI_IP
+#define PIM_MAX_BITLEN	IPV4_MAX_BITLEN
+
+union pimprefixptr {
+	prefixtype(pimprefixptr, struct prefix,      p)
+	prefixtype(pimprefixptr, struct prefix_ipv4, p4)
+} TRANSPARENT_UNION;
+
+union pimprefixconstptr {
+	prefixtype(pimprefixconstptr, const struct prefix,      p)
+	prefixtype(pimprefixconstptr, const struct prefix_ipv4, p4)
+} TRANSPARENT_UNION;
+
 #else
 typedef struct in6_addr pim_addr;
-#define PIM_ADDRSTRLEN INET6_ADDRSTRLEN
+
+#define PIM_ADDRSTRLEN	INET6_ADDRSTRLEN
+#define PIM_AF		AF_INET6
+#define PIM_AFI		AFI_IP6
+#define PIM_MAX_BITLEN	IPV6_MAX_BITLEN
+
+union pimprefixptr {
+	prefixtype(pimprefixptr, struct prefix,      p)
+	prefixtype(pimprefixptr, struct prefix_ipv6, p6)
+} TRANSPARENT_UNION;
+
+union pimprefixconstptr {
+	prefixtype(pimprefixconstptr, const struct prefix,      p)
+	prefixtype(pimprefixconstptr, const struct prefix_ipv6, p6)
+} TRANSPARENT_UNION;
 #endif
 
+/* for assignment/initialization (C99 compound literal)
+ * named PIMADDR_ANY (not PIM_ADDR_ANY) to match INADDR_ANY
+ */
+#define PIMADDR_ANY (pim_addr){ }
```

Comment:
Can we have PIMADDR_NONE (pim_addr)

Currently we are initialising ipv4 addresses to INADDR_NONE at some places

Reply:
I assume you mean renaming `PIMADDR_ANY` to `PIMADDR_NONE` ?

Reply:
PIMADDR_ANY serves the purpose of >>> #define INADDR_ANY              ((in_addr_t) 0x00000000)
We need PIMADDR_NONE which will serve the purpose of >>> #define INADDR_NONE             ((in_addr_t) 0xffffffff)

I will develop this, since this PR is already merged.